### PR TITLE
Migrate away from custom junit, jmock

### PR DESCRIPTION
**Pull request by niloc132**
This lets us dispose of the custom jars, and avoid java 16+ compatibility issues (at least when jacoco is disabled).

### CODE REVIEW

**Comment by niloc132** [comment]
See https://github.com/niloc132/deephaven-core/actions/runs/2391011627 for nightly build of the (as of writing) HEAD commit.

@rcaudy please review for the generics changes in tree stuff - We can go deeper here it seems and remove some of the other references to Client. Those generics don't seem to serve a purpose right now, and they were causing issues with making it possible to write the correct generics for this code:
```
                {
                    allowing(clients[myI]).addDisconnectHandler(
                            with(AbstractExpectations.<Consumer<TreeTableClientTableManager.Client>>anything()));
                    allowing(clients[myI]).removeDisconnectHandler(
                            with(AbstractExpectations.<Consumer<TreeTableClientTableManager.Client>>anything()));
                }
                ```

**Comment by rcaudy** [comment]
No concerns from my end.

File: buildSrc/src/main/groovy/io.deephaven.java-classpath-conventions.gradle [hunk 1]
```diff
@@ -61,7 +61,9 @@ dependencies {
     jama 'gov.nist.math.jama:gov.nist.math.jama:1.1.1'
     mockrunner 'com.mockrunner:mockrunner-jdbc:1.0.4'
     commonsCsv 'org.apache.commons:commons-csv:1.4'
-    junit fileTree(dir: "${rootDir}/test-libs", include: ['*.jar'])
+    junit 'junit:junit:4.13.2'
+    junit 'org.jmock:jmock-junit4:2.12.0'
+    testCompile 'org.jmock:jmock-imposters:2.12.0'
```
Comment (devinrsmith) [inline buildSrc/src/main/groovy/io.deephaven.java-classpath-conventions.gradle:66]: Was this meant to be junit?

**Review by devinrsmith — COMMENTED** [review]
Great work!

File: buildSrc/src/main/groovy/io.deephaven.java-classpath-conventions.gradle (hunk 1 above)
Comment (niloc132) [reply buildSrc/src/main/groovy/io.deephaven.java-classpath-conventions.gradle:66]: Ah yes, will fix.

**Review by devinrsmith — APPROVED** [review]

**Comment by niloc132** [comment]
All green for nightlies https://github.com/niloc132/deephaven-core/actions/runs/2397117552

**Pull request merged**